### PR TITLE
Test reporting improvements on CircleCI interface

### PR DESCRIPTION
**Pull request by evertonfraga**
The main reason for this PR is to quickly identify what are the failing tests, instead of skimming through the thousands of lines in the test output.

This PR makes some changes on how tests are ran, so we can take advantage of test analysis and reporting by circleCI. Lifecycle additions:

1. Test commands got a `| tee -a results.tap` suffix. It gets `stdout` and `stderr` from tests, print them out to CI log and save them to a file, which is required for the following step.
2. The package `tap-xunit` converts the original tap format to [xunit](https://xunit.net/docs/format-xml-v2).
3. CircleCI reads that xml and produce a nice output and feed the tests metadata to their Insights platform.

Related: 
- [tape reporters](https://github.com/substack/tape#pretty-reporters)
- [xunit format](https://xunit.net/docs/format-xml-v2)
- [expanded insights via circleCI](https://circleci.com/blog/how-to-output-junit-tests-through-circleci-2-0-for-expanded-insights/)

Should all tests pass, here's the summary:

Screen Shot 2019-11-14 at 11 52 33 AM

Failed tests show up like this ([live example](https://circleci.com/gh/evertonfraga/ethereumjs-vm/108)):

image

### CODE REVIEW

File: lib/state/stateManager.ts [hunk 1]
```diff
@@ -299,7 +299,7 @@ export default class StateManager {
         if (value && value.length) {
           // format input
           const encodedValue = encode(value)
-          storageTrie.put(key, encodedValue, done)
+          storageTrie.put(key, null, done)
```
Comment (evertonfraga) [inline lib/state/stateManager.ts]: This change is just to force tests to fail. It'll be reverted soon.

**Comment by holgerd77** [comment]
Just for your information: @danjm also experimented with test reporting some time ago and introduced some reporting improvements along the way, see [developer docs](https://github.com/ethereumjs/ethereumjs-vm/blob/master/developer.md#running-tests-with-a-reporterformatter) for reference.

**Comment by evertonfraga** [comment]
@holger77 sure, that documentation was my starting point. 

in short: the test format changes I made are for CircleCI internally only. I’ll lay down some other (less verbose) formatter options for us, humans.

**Comment by evertonfraga** [comment]
Ready for review. I squashed the commits, because the changes were really simple.

File: package.json [hunk 2]
```diff
@@ -63,6 +63,7 @@
     "merkle-patricia-tree": "^2.3.2",
     "rustbn.js": "~0.2.0",
     "safe-buffer": "^5.1.1",
+    "tap-xunit": "^2.4.1",
```
Comment (s1na) [inline package.json]: Is there a way we could have this as a `devDependency`?

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,23 +1,39 @@
...
+    store_test_results:
+      path: tests/tape/
+
 jobs:
   install:
     <<: *defaults
     steps:
       - checkout
       - *restore_node_modules
+      - run: npm install
```
Comment (s1na) [inline .circleci/config.yml]: Isn't this redundant?

File: package.json (hunk 2 above)
Comment (evertonfraga) [reply package.json]: absolutely, my mistake

File: .circleci/config.yml (hunk 3 above)
Comment (evertonfraga) [reply .circleci/config.yml]: It is indeed redundant. Squash issue, will fix. good catch!

**Comment by evertonfraga** [comment]
This PR can be re-assessed now.

**Comment by holgerd77** [comment]
Great that this has been merged! 😄

Just for my understanding and for confirmation: so when tests pass one is not seeing anything extra displayed by the new functionality on the CI output, this only comes into place on errors. Is this correct?

**Comment by evertonfraga** [comment]
@holgerd77 regarding the Test Summary tab, this is correct. Only failed tests are displayed in detail. 

Additionally, it’ll start to feed more information to the the Insights platform (found on the sidebar): most failed tests, etc.

**Comment by holgerd77** [comment]
@evertonfraga ah, thanks! 👍

**Pull request merged**